### ISSUE TITLE
Allow using generic git servers

### ISSUE THREAD

**Issue by mtb-xt** [argoproj-labs/argocd-autopilot#529]
Hi!

Currently, if I have a not supported git server implementation, I cannot use argocd-autopilot. 
Please add a feature to allow specifying a 'generic' git server type, with certain switches to avoid using the API.
(so I can tell it about master branch name and other required parameters manually).

It would be ideal, if autopilot had support for every git server out there, but it is unrealistic, so having a `generic` option seems a nice middle ground.

Another thing - certain git server implementations do not use `.git` suffix at the end of the repository path (for example - [onedev](https://onedev.io), so I have to create a repository literally called `something.git` to make it work. Please do not force add `.git` suffix at the end of the repository URL in a `generic` implementation.

**Comment by noam-codefresh** [comment]
just to understand the issue -
at the moment, autopilot does not do anything provider-specific in *most* cases, unless the user supplies a repoURL for a repository that does not exist (only in `repo bootstrap` command). in that case, there is a provider-specific api call to create that repository.

another edge case - if running inside a docker container (or anywhere else without a valid .gitconfig that defines the username and email of the committing user. autopilot will try to get the user info by their token - another provider-specific api call.

but your issue is in a different case - autopilot currently automatically adds `.git` to the repoURL (between the url and the path), and this is not compatible with ondev (for example)?

**Comment by mtb-xt** [comment]
> just to understand the issue - at the moment, autopilot does not do anything provider-specific in _most_ cases, unless the user supplies a repoURL for a repository that does not exist (only in `repo bootstrap` command). in that case, there is a provider-specific api call to create that repository.
Yes, this is correct - I was  thinking there should be a 'generic' provider, that implies there's no API action going under the hood.

> but your issue is in a different case - autopilot currently automatically adds `.git` to the repoURL (between the url and the path), and this is not compatible with ondev (for example)?

Yes, this is the main issue I have, actually. With all providers, autopilot forces the `.git` suffix to the repoURL (and this, combined with my previous thought, it may be solved by having a 'generic' provider, without any API magic or repoURL manipulation)
